contracts/exchange-forwarder/contracts/src/MixinForwarderCore.sol

```diff
@@ -47,164 +53,114 @@ contract MixinForwarderCore is
     function marketSellOrdersWithEth(
         LibOrder.Order[] memory orders,
         bytes[] memory signatures,
-        LibOrder.Order[] memory feeOrders,
-        bytes[] memory feeSignatures,
         uint256 feePercentage,
         address payable feeRecipient
     )
         public
         payable
-        returns (
-            FillResults memory orderFillResults,
-            FillResults memory feeOrderFillResults
-        )
+        returns (FillResults memory orderFillResults)
     {
         // Convert ETH to WETH.
         _convertEthToWeth();
 
-        uint256 wethSellAmount;
-        uint256 zrxBuyAmount;
-        uint256 makerAssetAmountPurchased;
-        if (orders[0].makerAssetData.equals(ZRX_ASSET_DATA)) {
-            // Calculate amount of WETH that won't be spent on ETH fees.
-            wethSellAmount = _getPartialAmountFloor(
-                PERCENTAGE_DENOMINATOR,
-                _safeAdd(PERCENTAGE_DENOMINATOR, feePercentage),
-                msg.value
-            );
-            // Market sell available WETH.
-            // ZRX fees are paid with this contract's balance.
-            orderFillResults = _marketSellWeth(
-                orders,
-                wethSellAmount,
-                signatures
-            );
-            // The fee amount must be deducted from the amount transfered back to sender.
-            makerAssetAmountPurchased = _safeSub(orderFillResults.makerAssetFilledAmount, orderFillResults.takerFeePaid);
-        } else {
-            // 5% of WETH is reserved for filling feeOrders and paying feeRecipient.
-            wethSellAmount = _getPartialAmountFloor(
-                MAX_WETH_FILL_PERCENTAGE,
-                PERCENTAGE_DENOMINATOR,
-                msg.value
-            );
-            // Market sell 95% of WETH.
-            // ZRX fees are payed with this contract's balance.
-            orderFillResults = _marketSellWeth(
-                orders,
-                wethSellAmount,
-                signatures
-            );
-            // Buy back all ZRX spent on fees.
-            zrxBuyAmount = orderFillResults.takerFeePaid;
-            feeOrderFillResults = _marketBuyExactZrxWithWeth(
-                feeOrders,
-                zrxBuyAmount,
-                feeSignatures
-            );
-            makerAssetAmountPurchased = orderFillResults.makerAssetFilledAmount;
-        }
-
-        // Transfer feePercentage of total ETH spent on primary orders to feeRecipient.
+        // Calculate amount of WETH that won't be spent on the forwarder fee.
+        uint256 wethSellAmount = _getPartialAmountFloor(
+            PERCENTAGE_DENOMINATOR,
+            _safeAdd(PERCENTAGE_DENOMINATOR, feePercentage),
+            msg.value
+        );
+
+        // Spends up to wethSellAmount to fill orders and pay WETH order fees.
+        uint256 wethSpentAmount;
+        uint256 makerAssetAcquiredAmount;
+        (
+            orderFillResults,
```

Comment:
It looks like we don't actually use `orderFillResults` at all. We can probably simplify the logic in `MixinExchangeWrapper` if we don't even track or return `totalFillResults`.